src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,30 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = ();
```

Comment:
I'd advise to leave these unimplemented entirely until the blanket impl comes-- otherwise the blanket impl will be a breaking change (changing the error type from your custom uninhabited type to `!`).